src/runtime_src/core/pcie/driver/linux/xocl/subdev/mb_scheduler.c

```diff
@@ -4495,6 +4495,30 @@ int cu_map_addr(struct platform_device *pdev, u32 cu_idx, void *drm_filp,
 	return 0;
 }
 
+static int
+hot_reset(struct platform_device *pdev)
+{
+	struct exec_core *exec = platform_get_drvdata(pdev);
+	struct xocl_dev	*xdev = xocl_get_xdev(pdev);
+
+	/*
+	 * reclaim outstanding cmds, exec_rest_cmds is responsible for
+	 * recycling all cmds without leak.
+	 */
+	exec_reset_cmds(exec);
+	atomic_set(&xdev->outstanding_execs, 0);
```

Comment:
Silently setting outstanding_execs to 0 is maybe not a good idea?    exec_reset_cmds should have decremented this number,   If the count is off (not zero) then there is some other bug,.  Could we at least log a message if this ever happens?